### PR TITLE
VP-6070: Complete profile authentication and authorization scenarios

### PR DESCRIPTION
**Pull request by AliveMen**
### Related task:     
VP-6070 Complete Profile authentication authorization scenarios

Updated authorization

### CODE REVIEW

File: src/XProfile/VirtoCommerce.XProfile/Schemas/ProfileSchema.cs [hunk 1]
```diff
@@ -366,51 +394,54 @@ firstName lastName
...
-            var signInManager = _signInManagerFactory();
-            var user = await signInManager.UserManager.FindByIdAsync(userId) ?? new ApplicationUser
+            if (!permissions.IsNullOrEmpty())
             {
-                Id = userId,
-                UserName = "Anonymous",
-            };
+                var signInManager = _signInManagerFactory();
+                var userid = userPrincipal.FindFirstValue("name");
+                var user = await signInManager.UserManager.FindByIdAsync(userid) ;
```
Comment (yecli) [inline src/XProfile/VirtoCommerce.XProfile/Schemas/ProfileSchema.cs:420]: Why do we need to create a principal one more time?

File: src/XProfile/VirtoCommerce.XProfile/Schemas/ProfileSchema.cs [hunk 2]
```diff
@@ -366,51 +394,54 @@ firstName lastName
...
 
                          return await _mediator.Send(command);
                      })
                      .FieldType);
         }
 
 
-
-        private async Task CheckAuthAsync(string userId, object resource, params string[] permissions)
+        private async Task CheckAuthAsync(ClaimsPrincipal userPrincipal, object resource, params string[] permissions)
```
Comment (yecli) [inline src/XProfile/VirtoCommerce.XProfile/Schemas/ProfileSchema.cs:413]: Seems that `ClaimsPrincipal` parameter is redundant, as imho in all places `context.GetCurrentUser()` is used. Correct me if I'm wrong. 
So I think we could remove this param.

File: src/XProfile/VirtoCommerce.XProfile/Schemas/ProfileSchema.cs [hunk 3]
```diff
@@ -366,51 +394,54 @@ firstName lastName
...
         {
-            if (userId == null)
-            {
-                throw new ExecutionError($"argument {nameof(userId)} is null");
-            }
 
-            var signInManager = _signInManagerFactory();
-            var user = await signInManager.UserManager.FindByIdAsync(userId) ?? new ApplicationUser
+            if (!permissions.IsNullOrEmpty())
```
Comment (yecli) [inline src/XProfile/VirtoCommerce.XProfile/Schemas/ProfileSchema.cs:416]: You could use instant return here if you are using it in other places like this
```csharp
if (permissions.IsNullOrEmpty()) 
{
    return;
}
```

**Review by yecli — APPROVED** [review]

**Comment by mvktsk** [comment]
Review task https://virtocommerce.atlassian.net/browse/VP-6243 has been created

**Review by tatarincev — APPROVED** [review]

**Pull request merged**